packages/flutter/lib/src/material/dialog.dart

```diff
@@ -244,7 +245,7 @@ class Dialog extends StatelessWidget {
             surfaceTintColor: surfaceTintColor ?? dialogTheme.surfaceTintColor ?? defaults.surfaceTintColor,
             shape: shape ?? dialogTheme.shape ?? defaults.shape!,
             type: MaterialType.card,
-            clipBehavior: clipBehavior,
+            clipBehavior: clipBehavior ?? dialogTheme.clipBehavior ?? defaults.clipBehavior ?? Clip.none,
```

Comment:
```suggestion
            clipBehavior: clipBehavior ?? dialogTheme.clipBehavior ??  defaults.clipBehavior!,
```
After updating M3 defaults, update like this.